In [ ]:
import pandas as pd

df = pd.read_csv('/content/GoEmotions.csv.csv')  # use your actual file name
print(df.head())

print("\nShape:", df.shape)
print("\nColumns:", df.columns)

                                                text       id  \
0                                    That game hurt.  eew5j0j   
1   >sexuality shouldn’t be a grouping category I...  eemcysk   
2     You do right, if you don't care then fuck 'em!  ed2mah1   
3                                 Man I love reddit.  eeibobj   
4  [NAME] was nowhere near them, he was by the Fa...  eda6yn6   

                author            subreddit    link_id   parent_id  \
0                Brdd9                  nrl  t3_ajis4z  t1_eew18eq   
1          TheGreen888     unpopularopinion  t3_ai4q37   t3_ai4q37   
2             Labalool          confessions  t3_abru74  t1_ed2m7g7   
3        MrsRobertshaw             facepalm  t3_ahulml   t3_ahulml   
4  American_Fascist713  starwarsspeculation  t3_ackt2f  t1_eda65q2   

    created_utc  rater_id  example_very_unclear  admiration  ...  love  \
0  1.548381e+09         1                 False           0  ...     0   
1  1.548084e+09        37               

In [ ]:
emotion_cols = df.columns[9:]  # all emotion columns
print(emotion_cols)

Index(['admiration', 'amusement', 'anger', 'annoyance', 'approval', 'caring',
       'confusion', 'curiosity', 'desire', 'disappointment', 'disapproval',
       'disgust', 'embarrassment', 'excitement', 'fear', 'gratitude', 'grief',
       'joy', 'love', 'nervousness', 'optimism', 'pride', 'realization',
       'relief', 'remorse', 'sadness', 'surprise', 'neutral'],
      dtype='object')


In [ ]:
df['label'] = df[emotion_cols].idxmax(axis=1)

In [ ]:
df = df[['text', 'label']]
print(df.head())

                                                text       label
0                                    That game hurt.     sadness
1   >sexuality shouldn’t be a grouping category I...  admiration
2     You do right, if you don't care then fuck 'em!     neutral
3                                 Man I love reddit.        love
4  [NAME] was nowhere near them, he was by the Fa...     neutral


In [ ]:
print(df['label'].value_counts())

label
neutral           1381
admiration         442
approval           399
annoyance          289
disapproval        220
gratitude          207
curiosity          206
anger              199
amusement          192
confusion          179
disappointment     152
love               134
excitement         133
realization        129
optimism           129
caring             123
joy                119
sadness             96
disgust             80
surprise            79
desire              68
fear                59
remorse             55
relief              34
embarrassment       32
nervousness         20
pride               11
grief               10
Name: count, dtype: int64


for researchPaper:

Which emotion is most frequent?

Which is least?

Is dataset balanced?
Example:

neutral → 1381 😳 (very high)

pride → 11 😬 (very low)

grief → 10

👉 This is called class imbalance
The dataset exhibits significant class imbalance, with the ‘neutral’ class dominating the distribution, while emotions such as ‘grief’ and ‘pride’ have very limited samples.

In [ ]:
counts = df['label'].value_counts()

valid_labels = counts[counts > 50].index

df = df[df['label'].isin(valid_labels)]

print(df['label'].value_counts())

label
neutral           1381
admiration         442
approval           399
annoyance          289
disapproval        220
gratitude          207
curiosity          206
anger              199
amusement          192
confusion          179
disappointment     152
love               134
excitement         133
optimism           129
realization        129
caring             123
joy                119
sadness             96
disgust             80
surprise            79
desire              68
fear                59
remorse             55
Name: count, dtype: int64


In [ ]:
df = df.reset_index(drop=True)

In [ ]:
import os

# Create the directory if it doesn't exist
output_dir = '../data'
if not os.path.exists(output_dir):
    os.makedirs(output_dir)

df.to_csv(os.path.join(output_dir, 'cleaned_data.csv'), index=False)

In [ ]:
df['label'].value_counts()

,count
label,
neutral,1381
admiration,442
approval,399
annoyance,289
disapproval,220
gratitude,207
curiosity,206
anger,199
amusement,192
